Calendar/Calendar.js

```diff
@@ -0,0 +1,292 @@
+      : moment().startOf(calendarData.formats[this.props.defaultView]),
+  }
+
+  dayCellDimension = {
+    width: (this.props.calWidth - (2 * this.props.calPadding)) / 7,
+    height: (this.props.calWidth - (2 * this.props.calPadding)) / 7,
+  }
+
+  cellDimension = {
+    width: (this.props.calWidth - (2 * this.props.calPadding)) / 4,
+    height: (this.props.calWidth - (2 * this.props.calPadding)) / 4,
+  }
+
+  render() {
+    const {
+      calWidth, calPadding, close,
+    } = this.props;
+
+    return (
+      <View style={[styles.container, { width: calWidth }]}>
+        <TouchableOpacity
+          style={[styles.calRow, { paddingTop: calPadding * 2, paddingBottom: calPadding * 2 }]}
+          onPress={close}
+        >
+          {this.renderToggleArrow('prev')}
+          <View style={styles.centerItems}>
+            {this.renderHeader()}
+          </View>
+          {this.renderToggleArrow('next')}
+        </TouchableOpacity>
+        <View style={{ padding: calPadding }}>
+          {this.state.activeView === 'day' ? this.renderWeekDays() : null}
+          {this.renderData()}
+        </View>
+      </View>
+    );
+  }
+
+  renderHeader = () => {
+    const { activeView, currentDate } = this.state;
+    switch (activeView) {
+      case 'day':
+        return (
+          <View style={{ flexDirection: 'row' }}>
+            <TouchableOpacity onPress={() => this.onViewChange('month')}>
+              <Text style={[styles.headerStyle, styles.headerDayMonth]}>
+                {currentDate.format('MMMM')}
+              </Text>
+            </TouchableOpacity>
+            <TouchableOpacity onPress={() => this.onViewChange('year')}>
+              <Text style={styles.headerStyle}>
+                {currentDate.format('YYYY')}
+              </Text>
+            </TouchableOpacity>
+          </View>
+        );
+      case 'month':
+        return (
+          <TouchableOpacity onPress={() => this.onViewChange('year')}>
+            <Text style={styles.headerStyle}>{currentDate.format('YYYY')}</Text>
+          </TouchableOpacity>
+        );
+      default:
+        return <TouchableOpacity onPress={this.props.close}><Text>Cancel</Text></TouchableOpacity>;
+    }
+  }
+
+  renderToggleArrow = (toggle) => {
+    const arrowStyle = toggle === 'next' ? { transform: [{ rotate: '180deg' }] } : {};
+    return (
+      <TouchableOpacity style={styles.centerItems} onPress={() => this.changeDate(toggle)}>
+        <Image source={Back} style={arrowStyle} />
+      </TouchableOpacity>
+    );
+  }
+
+  renderData = () => {
+    return this.state.activeView === 'day' ? this.renderDays() : this.renderYearsOrMonths();
+  }
+
+  renderDays = () => {
+    const { currentDate } = this.state;
+    const monthIndex = currentDate.month();
+    const day = currentDate.clone().startOf('month').startOf('week');
+    let done = false;
+    const dayDates = [];
+
+    while (!done) {
+      dayDates.push(this.renderWeeks(day.clone(), monthIndex));
```

Comment:
@SilwalUjjwal don't clone day as it will be changed and iterated so that you won't have to add 7 days like in below